setup.py

```diff
@@ -1,8 +1,15 @@
 """Setup script for lasio"""
 
+import fileinput
+import lasio
+import re
 from setuptools import setup
 import os
 
+lasio_version = lasio.version()
```

Comment:
I mean it should not just suddenly change from 0.30 to 0.40, it will require us updating this dependency and re-applying our custom patches

Reply:
I sort of struggle to see how it can be a problem. The only reason we know that lasio has a version is because it is implemented poorly. 
If we want to keep the ability get version from git at build time I would recommend just calling `setup()` with `version=lasio.version()` and `use_scm_version=False`. It is much cleaner than updating file in build time.

Reply:
ah, that would still fail, because version if called from the `__init__`.

Reply:
Yea, when a new background process is spawned it will re-import lasio and run `__init__` again.

Reply:
tbh I'm still not a fan of this approach. Hardcoded version still seems like easiest approach (yes, it won't retrieve post1 from git, but who cares?).
What if the upstream repository changes the content of `__init__.py` e.g. instead of `__version__ = version()` it will now have a line like `__version__ = get_version()`. With the current implementation `__version__` won't be replaced and the issue you are trying to fix will get back. How would the person who does the migration suppose to remember about this fix?
I would say at the very least this solution should have some error handling and it should abort the build if it has not found the string.

Reply:
I could make my replace less specific such as `__version__ = ...` (which was how my original regex worked), I replaced anything on the right of `__version__ = ` with it. I could check to see if the replacement was complete and then raise an exception if nothing has changed, failing the build (I think this is the best approach, personally, and add some documentation in setup.py to why we are doing it).

If I do hardcode the value to `__version__ = '0.30'` and then we upgrade to `0.40`, this would guarantee a merge conflict. I suppose that could be beneficial as it highlights that it needs addressing, but how would the person upgrading and addressing the conflict know of this case and realize that they must now replace it to `__version__ = '0.40'`, maybe they check the commits and read the case, still, more work.

Reply:
they would know this case by looking into git history or git annotate and they would have access to Jira. If you leave a comment in the `__init__.py` it will make it even easier for them.
I'm fine with either hardcoded version or more robust version of this hack.

Reply:
Aye that's true, didn't think about this line changing. From my point of view I still think dynamic is better, but yeah it would need to be more robust. We know `__version__` is not going to change, since that is pretty much a standard. But we need to handle the rest of the line changing

Reply:
Updated.